pages/dao/[symbol]/proposal/components/instructions/Raydium/AddLiquidityToPool.tsx

```diff
@@ -0,0 +1,261 @@
+import BigNumber from 'bignumber.js'
+import * as yup from 'yup'
+import { BN } from '@project-serum/anchor'
+import {
+  Governance,
+  ProgramAccount,
+  serializeInstructionToBase64,
+} from '@solana/spl-governance'
+import { PublicKey } from '@solana/web3.js'
+import Input from '@components/inputs/Input'
+import Select from '@components/inputs/Select'
+import useGovernedMultiTypeAccounts from '@hooks/useGovernedMultiTypeAccounts'
+import useRealm from '@hooks/useRealm'
+import { createAddLiquidityInstruction } from '@tools/sdk/raydium/createAddLiquidityInstruction'
+import {
+  getAmountOut,
+  getLiquidityPoolKeysByLabel,
+} from '@tools/sdk/raydium/helpers'
+import { liquidityPoolKeysList } from '@tools/sdk/raydium/poolKeys'
+import { debounce } from '@utils/debounce'
+import { isFormValid } from '@utils/formValidation'
+import {
+  AddLiquidityRaydiumForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+import useWalletStore from 'stores/useWalletStore'
+
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+
+const AddLiquidityToPool = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const { realmInfo } = useRealm()
+  const { governedMultiTypeAccounts } = useGovernedMultiTypeAccounts()
+
+  const shouldBeGoverned = index !== 0 && governance
+  const programId: PublicKey | undefined = realmInfo?.programId
+  const [form, setForm] = useState<AddLiquidityRaydiumForm>({
+    governedAccount: undefined,
+    liquidityPool: '',
+    baseAmountIn: 0,
+    quoteAmountIn: 0,
+    fixedSide: 'base',
+    slippage: 0.5,
+  })
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
+    setFormErrors(validationErrors)
+    return isValid
+  }
+  async function getInstruction(): Promise<UiInstruction> {
+    const isValid = await validateInstruction()
+    let serializedInstruction = ''
+    if (
+      isValid &&
+      programId &&
+      form.governedAccount?.governance?.account &&
+      wallet?.publicKey
+    ) {
+      const poolKeys = getLiquidityPoolKeysByLabel(form.liquidityPool)
+      const [base, quote] = await Promise.all([
+        connection.current.getTokenSupply(poolKeys.baseMint),
+        connection.current.getTokenSupply(poolKeys.quoteMint),
+      ])
+
+      const createIx = createAddLiquidityInstruction(
+        poolKeys,
+        new BN(
+          new BigNumber(form.baseAmountIn.toString())
```

Comment:
FYI: I do apply a naming convention about amounts:

- If the amount is in native format, I add the prefix `native` on the variable name, like: `nativeBaseAmountIn`.
- If the amount is in ui format, I add the prefix `ui` on the variable name, like: `uiBaseAmountIn`.

it helps me knowing at all time what value I'm manipulating.